ezyrb/regular_grid.py

```diff
@@ -0,0 +1,131 @@
+
+        if np.prod(nN) != len(vals_scrmbld):
+            raise ValueError("points and values are not on a regular grid")
+        new_row_index = calculate_flat_index(iN, nN)
+        reverse_scrambling = np.argsort(new_row_index)
+        vals_on_regular_grid = vals_scrmbld[reverse_scrambling, :]
+        return grid_axes, vals_on_regular_grid
+
+    def fit(self, points, values, **kvargs):
+        """
+        Construct the interpolator given `points` and `values`.
+        Assumes that the points are on a regular grid, fails when not.
+        see scipy.interpolate.RegularGridInterpolator
+
+        :param array_like points: the coordinates of the points.
+        :param array_like values: the values in the points.
+        """
+        # we have two options
+        # 1.: we could make an interpolator for every mode and its coefficients
+        # or 2.: we "interpolate" the mode number
+        # option 1 is cleaner, but option 2 performs better
+        # X = U S VT, X being shaped (m, n)
+        points = np.array(points)
+        if not np.issubdtype(points.dtype, np.number):
+            raise ValueError('Invalid format or dimension for the argument'
+                             '`points`.')
+
+        if len(points.shape) == 1:
+            points.shape = (-1, 1)
+
+        self.dim = len(points[0])
+        vals = np.asarray(values)
+        grid_axes, values_grd = self.get_grid_axes(points, vals)
+        self.n_modes = vals.T.shape[0]  # vals = (S@VT).T = S@V
+        if self.n_modes > 1:
+            self.mode_nr = np.arange(self.n_modes)
+            extended_grid = [self.mode_nr, *grid_axes]
+            shape = [self.n_modes, ]
+        else:
+            extended_grid = grid_axes
+            shape = []
+        for i in range(self.dim):
+            shape.append(len(grid_axes[i]))
+        assert np.prod(shape) == values_grd.size, "Values don't match grid. "\
+            "Make sure to pass a grid, not a list of points!\n"\
+            "HINT: did you use rom.fit()? This method does not work with a "\
+            "grid. Use reduction.fit(...) and approximation.fit(...) instead."
+        self.interpolator = RegularGridInterpolator(extended_grid,
+                                                    values_grd.T.reshape(
+                                                        shape),
+                                                    **kvargs)
+
+    def predict(self, new_point):
+        new_point = np.array(new_point)
+        if len(new_point.shape) == 1:
+            new_point.shape = (-1, 1)
+
+        dim = self.dim
+        if self.n_modes > 1:
+            xi_extended = np.zeros((len(self.mode_nr), len(new_point), dim+1))
+            xi_extended[:, :, 0] = self.mode_nr[:, None]
+            for i in range(dim):
+                xi_extended[:, :, i+1] = np.array(new_point)[:, i]
+        else:
+            xi_extended = new_point
+        return self.interpolator(xi_extended).T
+
+
```

Comment:
I suggest moving this method inside the class. Make it private with an underscore as first character, and add the @staticmethod decorator on top of it.

Reply:
I removed it entirely since it is implemented in numpy already... Thanks @fandreuz for the hint